client/js/_author/selectors/bank_navigator.js

```diff
@@ -0,0 +1,49 @@
+import _ from 'lodash';
+import { createSelector }  from 'reselect';
+import * as common from './common';
+import { _assessment } from './assessment';
+
+export const path = state => state.bankNavigation.location;
+
+
+export const currentBankId = createSelector(
+  path,
+  (_path) => {
+    if (!_.isEmpty(_path)) return _.last(_path).id;
+    return null;
+  }
+);
+
+export const rawAssessments = createSelector(
+  common.assessments,
+  currentBankId,
+  (_rawAssessments, _currentBankId) => _.get(_rawAssessments, _currentBankId, {})
+);
+
+export const bankAssessments = createSelector(
+rawAssessments,
+common.settings,
+(_rawAssessments, _settings) => {
+  const transformedAssessments =
```

Comment:
This is basically deserializing right? One day (not this day) we should do this in middleware.